### ISSUE TITLE
eb -S fails on things with "++" in it.

### ISSUE THREAD

**Issue by akesandgren** [easybuilders/easybuild-framework#2966]
```
eb -S netCDF-C++4-4.3.0-intel-2017a.eb
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/local/eb/software/Core/EasyBuild/3.9.1/lib/python2.7/site-packages/easybuild_framework-3.9.1-py2.7.egg/easybuild/main.py", line 457, in <module>
    main()
  File "/local/eb/software/Core/EasyBuild/3.9.1/lib/python2.7/site-packages/easybuild_framework-3.9.1-py2.7.egg/easybuild/main.py", line 224, in main
    terse=options.terse)
  File "/local/eb/software/Core/EasyBuild/3.9.1/lib/python2.7/site-packages/easybuild_framework-3.9.1-py2.7.egg/easybuild/tools/robot.py", line 486, in search_easyconfigs
    silent=True, filename_only=False)
  File "/local/eb/software/Core/EasyBuild/3.9.1/lib/python2.7/site-packages/easybuild_framework-3.9.1-py2.7.egg/easybuild/tools/filetools.py", line 616, in search_file
    query = re.compile(query, re.I)
  File "/usr/lib/python2.7/re.py", line 194, in compile
    return _compile(pattern, flags)
  File "/usr/lib/python2.7/re.py", line 251, in _compile
    raise error, v # invalid expression
sre_constants.error: multiple repeat
```

**Comment by boegel** [comment]
@akesandgren Can you try with the latest EasyBuild 3.9.3? I don't think that'll make much difference, but just to be sure this isn't a bug we've fixed already...

**Comment by boegel** [comment]
Nevermind, trivial to reproduce locally even with `develop`.

I'll take a look at fixing it...

**Comment by boegel** [comment]
fixed in #2967

I had to get creative to avoid breaking genuine queries that include special characters, like `^netCDF` or `n.*CDF`...
